etl/src/error.rs

```diff
@@ -0,0 +1,898 @@
+
+/// Converts [`sqlx::Error`] to [`ETLError`] with appropriate error kind.
+///
+/// Maps database errors to [`ErrorKind::QueryFailed`], I/O errors to [`ErrorKind::IoError`],
+/// and connection pool errors to [`ErrorKind::ConnectionFailed`].
+impl From<sqlx::Error> for ETLError {
+    fn from(err: sqlx::Error) -> ETLError {
+        let kind = match &err {
+            sqlx::Error::Database(_) => ErrorKind::QueryFailed,
+            sqlx::Error::Io(_) => ErrorKind::IoError,
+            sqlx::Error::PoolClosed | sqlx::Error::PoolTimedOut => ErrorKind::ConnectionFailed,
+            _ => ErrorKind::QueryFailed,
+        };
+
+        ETLError {
+            repr: ErrorRepr::WithDescriptionAndDetail(
+                kind,
+                "Database operation failed",
+                err.to_string(),
+            ),
+        }
+    }
+}
+
+// BigQuery error conversions (feature-gated)
+
+/// Converts [`gcp_bigquery_client::error::BQError`] to [`ETLError`] with appropriate error kind.
+///
+/// Maps errors based on their specific type for better error classification and handling.
+#[cfg(feature = "bigquery")]
+impl From<gcp_bigquery_client::error::BQError> for ETLError {
+    fn from(err: gcp_bigquery_client::error::BQError) -> ETLError {
+        use gcp_bigquery_client::error::BQError;
+
+        let (kind, description) = match &err {
+            // Authentication related errors
+            BQError::InvalidServiceAccountKey(_) => (
+                ErrorKind::AuthenticationError,
+                "Invalid BigQuery service account key",
+            ),
+            BQError::InvalidServiceAccountAuthenticator(_) => (
+                ErrorKind::AuthenticationError,
+                "Invalid BigQuery service account authenticator",
+            ),
+            BQError::InvalidInstalledFlowAuthenticator(_) => (
+                ErrorKind::AuthenticationError,
+                "Invalid BigQuery installed flow authenticator",
+            ),
+            BQError::InvalidApplicationDefaultCredentialsAuthenticator(_) => (
+                ErrorKind::AuthenticationError,
+                "Invalid BigQuery application default credentials",
+            ),
+            BQError::InvalidAuthorizedUserAuthenticator(_) => (
+                ErrorKind::AuthenticationError,
+                "Invalid BigQuery authorized user authenticator",
+            ),
+            BQError::AuthError(_) => (
+                ErrorKind::AuthenticationError,
+                "BigQuery authentication error",
+            ),
+            BQError::YupAuthError(_) => (
+                ErrorKind::AuthenticationError,
+                "BigQuery OAuth authentication error",
+            ),
+            BQError::NoToken => (
+                ErrorKind::AuthenticationError,
+                "BigQuery authentication token missing",
+            ),
+
+            // Network and transport errors
+            BQError::RequestError(_) => (ErrorKind::IoError, "BigQuery request failed"),
```

Comment:
We are swallowing details of the inner error here which will make it hard to debug errors when they occur in real pipelines. Instead of a detailed description of why a request failed we'll get a generic "BigQuery request failed". Can we keep the inner details intact to help debug issues.

Reply:
This is done via the `err.to_string()` at the bottom.

Reply:
Ah, ok missed that.